Dockerfile

```diff
@@ -1,31 +1,45 @@
-FROM quay.io/opendatahub/text-generation-inference:stable-bafd218
-
-USER root
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest as base
 
 # Add grpc-ecosystem health probe
 ARG GRPC_HEALTH_PROBE_VERSION=v0.4.20
 
-WORKDIR /caikit
-COPY caikit /caikit
-
-RUN yum -y update && yum -y install git git-lfs && yum clean all && \
-    git lfs install && \
-    pip install 'micropipenv[toml]' && \
-    micropipenv install && \
-    rm -rf ~/.cache && \
-    mkdir -p /opt/models && \
-    adduser -g 0 -u 1001 caikit --home-dir /caikit && \
-    chown -R 1001:0 /caikit /opt/models && \
-    chmod -R g=u /caikit /opt/models
-
 # This is for the use-cases without kserve
 RUN curl -Lo /usr/local/bin/grpc_health_probe https://github.com/grpc-ecosystem/grpc-health-probe/releases/download/${GRPC_HEALTH_PROBE_VERSION}/grpc_health_probe-linux-amd64 && \
     chmod +x /usr/local/bin/grpc_health_probe
 
-USER 1001
+RUN microdnf -y update && \
+    microdnf -y install \
+        shadow-utils git git-lfs python39-pip python39-wheel && \
+    git lfs install && \
+    pip3 install --no-cache-dir --upgrade pip wheel && \
+    microdnf clean all
+
+
+FROM base as poetry-builder
+ENV POETRY_VIRTUALENVS_IN_PROJECT=1
+
+WORKDIR /tmp/poetry
+COPY pyproject.toml .
+COPY poetry.lock .
+RUN pip3 install poetry && poetry install
+
+
+FROM base as deploy
+WORKDIR /caikit
+
+COPY --from=poetry-builder /tmp/poetry/.venv /caikit/
+
+ENV VIRTUAL_ENV=/caikit
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+
+RUN groupadd --system caikit --gid 1001 && \
+    adduser --system --uid 1001 --gid 0 --groups caikit \
+    --create-home --home-dir /caikit --shell /sbin/nologin \
+    --comment "Caikit User" caikit
+
+USER caikit
 
-ENV TRANSFORMERS_CACHE="/tmp/transformers_cache" \
-    RUNTIME_LIBRARY='caikit_nlp' \
-    RUNTIME_LOCAL_MODELS_DIR='/opt/models'
+ENV CONFIG_FILES=/caikit/config/caikit.yml
+VOLUME ["/caikit/config/"]
 
-CMD [ "./start-serving.sh" ]
+CMD ["python",  "-m", "caikit.runtime.grpc_server"]
```

Comment:
Let's also see about moving to the combined HTTP+gRPC entrypoint Gabe was talking about here, there should be a caikit.runtime entrypoint that launches both in recent caikit (may need to bump version to see it)

Reply:
How about doing this in another PR?

Reply:
Sure, that works, we'll need that one as next priority though so we have time to play with the HTTP API